courses-app/src/app/shared/course-form/course-form.component.html

```diff
@@ -0,0 +1,65 @@
+<form class="form-wrapper" #ngFormCourse="ngForm" [formGroup]="formGroupCourse" (ngSubmit)="onFormSubmit()">
+  <div class="title-wrapper">
+    <div class="input-wrapper">
+      <label class="label">Title
+      <input type="text" formControlName="title" required class="input" placeholder="Enter title..." >
+      <div *ngIf="title?.invalid && (ngFormCourse.submitted || title?.touched)"
+           class="alert alert-danger">
+        <div *ngIf="title?.errors?.['required']" class="invalid-feedback">
+          Name is required.
+        </div>
+      </div>
+      </label>
+    </div>
+    <app-button [text]="'Create course'"></app-button>
+  </div>
+  <label class="label">Description
+    <textarea type="text" formControlName="description" required class="textarea" placeholder="Enter description..."></textarea>
+    <div *ngIf="description?.invalid && (ngFormCourse.submitted || description?.touched)"
+         class="alert alert-danger">
+      <div *ngIf="description?.errors?.['required']" class="invalid-feedback">
+        Name is required.
+      </div>
+    </div>
+  </label>
+  <form class="forms-container" #ngFormAuthor="ngForm" [formGroup]="formGroupAuthor" (ngSubmit)="onAuthorFormSubmit()">
+    <div>
+      <p class="title">Add authors</p>
+      <div class="input-wrapper">
+        <label class="label">Author name</label>
+        <input class="input-grid" formControlName="authorName" type="text" >
+        <div *ngIf="authorName?.invalid && (ngFormAuthor.submitted || authorName?.touched)"
+             class="alert alert-danger">
+          <div *ngIf="authorName?.errors?.['required']" class="invalid-feedback">
+            Name is required.
+          </div>
+        </div>
+      </div>
+      <app-button [text]="'Create Author'"></app-button>
+    </div>
+    <div>
+      <p class="title">Author list</p>
+      <div class="author-list">
+        <ng-container *ngFor="let author of authors;index as i">
+          <p class="author">{{author.value}}</p>
+          <app-button [text]="'Delete author'" (click)="deleteAuthor(i)"></app-button>
+        </ng-container>
+      </div>
+    </div>
+    <div>
+      <p class="title">Duration</p>
+      <div class="input-wrapper">
+        <label class="label">Duration</label>
+        <input type="number" class="input-grid" formControlName="duration" required (input)="inputHandler($event)">
+        <div *ngIf="duration?.invalid && (ngFormCourse.submitted || duration?.touched)"
+             class="alert alert-danger">
+          <div *ngIf="duration?.errors?.['required']" class="invalid-feedback">
+            Name is required.
+          </div>
+        </div>
+      </div>
+      <span class="text">Duration {{duration ?? 0 | durationToTime}}</span>
+    </div>
+    <div></div>
+  </form>
```

Comment:
These forms should be reactive, but there is ngForm which is used in template-driven forms.